packages/back-end/src/integrations/SqlIntegration.ts

```diff
@@ -1518,6 +1557,19 @@ export default abstract class SqlIntegration
     return metric.queryFormat || (metric.sql ? "sql" : "builder");
   }
 
+  getDateTable(dateArray: string[]): string {
+    const dateString = dateArray
+      .map((d) => `SELECT ${d} AS day`)
+      .join("\nUNION ALL\n");
+    return `
+      SELECT ${this.dateTrunc(this.castToDate("t.day"))} AS day
```

Comment:
This may appear a little redundant, but there were some cases where directly applying `dateTrunc` was causing some problems (unfortunately I forget which engines this was originally true for).